packages/expo-modules-autolinking/src/ExpoModuleConfig.ts

```diff
@@ -16,8 +16,9 @@ export class ExpoModuleConfig {
   /**
    * Returns a list of names of Swift native modules classes to put to the generated modules provider file.
    */
-  iosModulesClassNames() {
-    return this.rawConfig.ios?.modulesClassNames ?? [];
+  iosModules() {
+    const iosConfig = this.rawConfig.ios;
+    return iosConfig?.modules ?? iosConfig?.modulesClassNames ?? [];
```

Comment:
Maybe add some comment here about `modulesClassNames` existence?
I suggest adding explanation and removal estimation.
```suggestion
    // 'modulesClassNames' is a legacy way to access registered native modules
    // TODO: to be removed <when>?
    return iosConfig?.modules ?? iosConfig?.modulesClassNames ?? [];
```